### PR TITLE
Mercury new-transaction

### PR DESCRIPTION
**Pull request by michelle0927**

### CODE REVIEW

File: components/mercury/mercury.app.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    _getBaseURL() {
+      return "https://backend.mercury.com/api/v1";
+    },
+    _getHeaders() {
+      return {
+        Authorization: `Bearer ${this.$auth.api_key}`,
+        "Content-Type": "application/json",
+      };
+    },
+    async _makeGetRequest(endpoint, params = null) {
```
Comment (dylburger) [inline components/mercury/mercury.app.js:16]: Can you abstract this method to accept the HTTP method and rename `_makeRequest`? That way it'll support `POST` requests, etc. naturally as we evolve the app file.

File: components/mercury/mercury.app.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      const config = {
+        method: "GET",
+        url: `${this._getBaseURL()}${endpoint}`,
+        headers: this._getHeaders(),
+        params,
+      };
+      return (await axios(config)).data;
+    },
+    dayAgo() {
```
Comment (dylburger) [inline components/mercury/mercury.app.js:25]: Similar feedback: maybe rename to `daysAgo` and pass the the number of days as an arg? Or rename to `oneDayAgo()` to make the return value explicit?

File: components/mercury/mercury.app.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      const dayAgo = new Date();
+      dayAgo.setDate(dayAgo.getDate() - 1);
+      return dayAgo;
+    },
+    async getAccounts() {
+      return (await this._makeGetRequest("/accounts")).accounts;
+    },
+    async getTransactions(accountId, params, limit, offset) {
```
Comment (dylburger) [inline components/mercury/mercury.app.js:33]: `limit` and `offset` aren't used - did you mean to pass them?

File: components/mercury/sources/new-transaction/new-transaction.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  props: {
+    mercury,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    account: {
```
Comment (dylburger) [inline components/mercury/sources/new-transaction/new-transaction.js:18]: Does it make sense to move this to the app file to support other sources in the future?

File: components/mercury/sources/new-transaction/new-transaction.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      type: "string",
+      label: "Account",
+      optional: false,
```
Comment (dylburger) [inline components/mercury/sources/new-transaction/new-transaction.js:21]: This is the default, so you can remove

File: components/mercury/sources/new-transaction/new-transaction.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      async options() {
```
Comment (dylburger) [inline components/mercury/sources/new-transaction/new-transaction.js:22]: Does this API support pagination?

File: components/mercury/sources/new-transaction/new-transaction.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      },
+    },
+  },
+  methods: {
+    getMeta(transaction) {
+      const { id, counterpartyName, postedAt } = transaction;
+      const postedAtDate = new Date(postedAt);
+      const ts = postedAtDate.getTime();
+      const summary = counterpartyName;
+      return { id, summary, ts };
```
Comment (dylburger) [inline components/mercury/sources/new-transaction/new-transaction.js:38]: You can drop a couple of lines by doing something like:

```javascript
const { id, counterpartyName: summary, postedAt } = transaction;
const ts = (new Date(postedAt)).getTime();
return { id, summary, ts };
```

**Review by dylburger — CHANGES_REQUESTED** [review]
Mostly small / style changes

File: components/mercury/sources/new-transaction/new-transaction.js (hunk 6 above)
Comment (michelle0927) [reply components/mercury/sources/new-transaction/new-transaction.js:22]: It does for listing transactions, but not for listing accounts. 
https://docs.mercury.com/reference#accounts

**Review by dylburger — APPROVED** [review]
lgtm!

**Pull request merged**